File: dd demo/src/main/java/andrew/volostnykh/dddemo/model/Detail.java
package andrew.volostnykh.dddemo.model;

import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.databind.JsonNode;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.Table;
import jakarta.persistence.Transient;
import lombok.Data;
import org.hibernate.annotations.JdbcTypeCode;
import org.hibernate.annotations.Type;
import org.hibernate.type.SqlTypes;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.time.temporal.ChronoUnit;
import java.util.List;
import java.util.Map;

@Data
@Entity
@Table(name = "details")
public class Detail {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String code;
    private String name;

    private String value;
    @Enumerated(EnumType.STRING)
    private DetailValueType valueType;

    private LocalDateTime date;
    @Enumerated(EnumType.STRING)
    private ChronoUnit chronoUnit;

    private String formula;

    @JdbcTypeCode(SqlTypes.JSON)
    private Map<LocalDateTime, BigDecimal> periodicData;

    @Transient
    @JsonIgnore
    private transient List<String> stackLog;
}

File: dd-demo-fe/src/app/model/detail.ts
export class Detail {
  id: number;
  code: string;
  name: string;
  value: string | number | null;
  valueType: string;
  date: Date | null;
  chronoUnit: string | null;
  formula: string | null;

  constructor(id: number,
              name: string,
              code: string,
              value: string | number | null,
              valueType: string,
              date: Date | null,
              chronoUnit: string | null,
              formula: string | null) {
    this.id = id;
    this.name = name;
    this.code = code;
    this.value = value;
    this.valueType = valueType;
    this.date = date;
    this.chronoUnit = chronoUnit;
    this.formula = formula;
  }

}
